bench/workgen/workgen_int.h

```diff
@@ -55,20 +55,24 @@ struct WorkgenTimeStamp {
     WorkgenTimeStamp() {}
 
     static uint64_t get_timestamp_lag(double seconds) {
-        uint64_t start_time;
-        workgen_clock(&start_time);
-
-        return (ns_to_us(start_time) - secs_us(seconds));
+        return (get_timestamp() - secs_us(seconds));
```

Comment:
This function does not seem clear to me. Given the definition of  `get_timestamp`, we get the epoch time and remove the given value to it? What if the given value is greater than the derived epoch time?

Reply:
That is how I found this was the function that was the issue (secs_us(seconds) > timestamp). This function's role is to get a timestamp that lags the current timestamp by an amount of seconds. Since it has switched to using epoch, it is unlikely to ever be smaller again.

Reply:
Ok, I see, thanks for explaining. We should probably ensure the required lag makes sense.